File: sqlitemagic-tests/app/src/main/java/com/siimkinks/sqlitemagic/TestApp.java
package com.siimkinks.sqlitemagic;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.sqlite.db.framework.FrameworkSQLiteOpenHelperFactory;

import com.siimkinks.sqlitemagic.annotation.Database;

import io.reactivex.schedulers.Schedulers;

@Database(
    name = BuildConfig.DB_NAME,
    version = BuildConfig.DB_VERSION
)
public class TestApp extends Application {
  public static Application INSTANCE;

  @Override
  public void onCreate() {
    super.onCreate();
    INSTANCE = this;
    deleteDatabase(SqlUtil.getDbName());
    initDb(this);
  }

  public static void initDb(@NonNull Application app) {
    SqliteMagic.setLoggingEnabled(true);
    SqliteMagic.builder(app)
        .sqliteFactory(new FrameworkSQLiteOpenHelperFactory())
        .scheduleRxQueriesOn(Schedulers.trampoline())
        .openDefaultConnection();
  }
}

File: sqlitemagic-tests/app-kotlin/src/main/kotlin/com/siimkinks/sqlitemagic/TestApp.kt
package com.siimkinks.sqlitemagic

import android.app.Application
import androidx.sqlite.db.framework.FrameworkSQLiteOpenHelperFactory
import io.reactivex.schedulers.Schedulers

class TestApp : Application() {
  lateinit var INSTANCE: Application

  override fun onCreate() {
    super.onCreate()
    INSTANCE = this
    initDb(this)
  }

  fun initDb(app: Application) {
    SqliteMagic.setLoggingEnabled(true)
    SqliteMagic.builder(app)
      .database(SqliteMagicDatabase())
      .sqliteFactory(FrameworkSQLiteOpenHelperFactory())
      .scheduleRxQueriesOn(Schedulers.trampoline())
      .openDefaultConnection()
  }
}